### PR TITLE
Sprint 4

### PR DESCRIPTION
**Pull request by Mikmist**
We have added the specific comments, changed the architecture document to fit new changes and added front and backend. We have not yet started working on extra code for the current sprint because of the exams that were last week.

### CODE REVIEW

File: Hestia/Hestia/DevicesScreen/AddDeviceScreen/TableSourceAddDeviceDevice.cs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            this.plugins = plugins;
+            this.owner = owner;
+            this.collection = collection;
+
+        }
+
+        // We have only one section with devices (thus far)
+        public override nint NumberOfSections(UITableView tableView)
+        {
+            return 1;
```
Comment (Feikoritsema) [inline Hestia/Hestia/DevicesScreen/AddDeviceScreen/TableSourceAddDeviceDevice.cs]: If the number of sections is hardcoded, do it in some config file or constant, don't let a function always return 1.

File: Hestia/Hestia/DevicesScreen/AddDeviceScreen/PropertyCell.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+{
+    public class PropertyCell : UITableViewCell
+    {
+        public UITextField inputField;
+        public PropertyCell(NSString cellId) : base(UITableViewCellStyle.Default, cellId)
+        {
+            // Create inputfield and add it to the cell
+            inputField = new UITextField();
+            ContentView.AddSubviews(new UIView[] {inputField});
+
```
Comment (Feikoritsema) [inline Hestia/Hestia/DevicesScreen/AddDeviceScreen/PropertyCell.cs]: Try to keep formatting consistent and neat, no double whittling after functions, and no single at the end.

File: Hestia/Hestia/DevicesScreen/AddDeviceScreen/TableSourceAddDeviceManufacturer.cs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        {
+            this.plugins = plugins;
+            this.owner = owner;
+            this.collections = collections;
+        }
+
+        // We have only one section with manufacturers
+        public override nint NumberOfSections(UITableView tableView)
+        {
+            return 1;
```
Comment (Feikoritsema) [inline Hestia/Hestia/DevicesScreen/AddDeviceScreen/TableSourceAddDeviceManufacturer.cs]: Same, if the number of sections changes you have to change all files, and it isn't neat to do this.

File: Hestia/Hestia/DevicesScreen/EditDevice/UIViewControllerEditDeviceName.cs [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+            newName.Frame = new CGRect(15, 10, 100, 31);
+            rectangle.AddSubview(newName);
+
+            // Inputfield
+            UITextField changeNameField = new UITextField();
+            changeNameField.Frame = new CGRect(110, 10, this.View.Bounds.Width - 125, 31);
+            changeNameField.Placeholder = device.Name;
+            rectangle.AddSubview(changeNameField);
+
+            this.View.BackgroundColor = UIColor.FromWhiteAlpha(0.95f, 1.0f);
```
Comment (Feikoritsema) [inline Hestia/Hestia/DevicesScreen/EditDevice/UIViewControllerEditDeviceName.cs]: Colors and other elements to some constants ?

File: Hestia/Hestia/DevicesScreen/TableSource.cs [hunk 5]
```diff
@@ -182,7 +174,7 @@ public void WillBeginTableEditing(UITableView tableView)
             // This should not be permanently stored, but trigger the add new
             // device screen on touch
             List<backend.models.Activator> temp_activator = new List<backend.models.Activator>();
-            NetworkHandler temp_networkhandler = new NetworkHandler("94.212.164.28", 8000);
+            NetworkHandler temp_networkhandler = new NetworkHandler(Globals.IP, Globals.Port);
```
Comment (Feikoritsema) [inline Hestia/Hestia/DevicesScreen/TableSource.cs]: Thanks, though will you make sure to ask the user for its own input IP first time ?

File: Hestia/Hestia/DevicesScreen/UITableViewControllerServerConnect.cs [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        public UITableViewControllerServerConnect(IntPtr handle) : base(handle)
+        {
+            
+        }
+
+        public override void ViewDidLoad()
+        {
+
+            base.ViewDidLoad();
+            newServerName.Text = "Hestia_Server2018";
```
Comment (Feikoritsema) [inline Hestia/Hestia/DevicesScreen/UITableViewControllerServerConnect.cs]: Why are these still here like this?

File: Hestia/Hestia/DevicesScreen/UITableViewControllerServerConnect.cs [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+            newIP.Text = "94.212.164.28";
+            newPort.Text = "8000";
+
+        }
+
+		public override bool ShouldPerformSegue(string segueIdentifier, NSObject sender)
+		{
+            if (segueIdentifier == "ServerToDevices")
+            {
+                if (newIP.Text == "94.212.164.28" && newPort.Text == "8000")
```
Comment (Feikoritsema) [inline Hestia/Hestia/DevicesScreen/UITableViewControllerServerConnect.cs]: So you only update if the server is the exact same as the old one? 
Why can't we change it, and you might user regex to check the IP input, and port range validation.

File: Hestia/Hestia/DevicesScreen/AddDeviceScreen/PropertyCell.cs [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+        public void UpdateCell(string placeholder)
+        {
+            inputField.Placeholder = placeholder;
+                       
+        }
+
+        public override void LayoutSubviews()
+        {
+           base.LayoutSubviews();
+           inputField.Frame = new CGRect(15, 4, ContentView.Bounds.Width - 30, ContentView.Bounds.Height - 8);
```
Comment (Feikoritsema) [inline Hestia/Hestia/DevicesScreen/AddDeviceScreen/PropertyCell.cs]: While it should work properly, defining all the exact locations and sizes like this isn't convenient and nice for future changes.

**Comment by Mikmist** [comment]
I think all of these changes have been made, I will close this PR and create a new one later tonight for sprint 5.

**Pull request closed without merging**